src/machines/modelingMachine.ts

```diff
@@ -736,6 +748,64 @@ export const modelingMachine = setup({
       // eslint-disable-next-line @typescript-eslint/no-floating-promises
       codeManager.updateEditorWithAstAndWriteToFile(kclManager.ast)
     },
+    'AST offset plane': ({ event }) => {
+      if (event.type !== 'Offset plane') return
+      if (!event.data) return
+      ;(async () => {
+        // Extract inputs
+        const ast = kclManager.ast
+        const { plane: selection, distance } = event.data
+
+        // Extract the default plane from selection
+        const plane = selection.otherSelections[0]
+        if (!(plane && plane instanceof Object && 'name' in plane))
+          return trap('No plane selected')
+
+        // Insert the distance variable if it exists
+        if (
+          'variableName' in distance &&
+          distance.variableName &&
+          distance.insertIndex !== undefined
+        ) {
+          const newBody = [...ast.body]
+          newBody.splice(
+            distance.insertIndex,
+            0,
+            distance.variableDeclarationAst
+          )
+          ast.body = newBody
+        }
+
+        // Get the default plane name from the selection
+
+        const offsetPlaneResult = addOffsetPlane({
+          node: ast,
+          defaultPlane: plane.name,
```

Comment:
![image](https://github.com/user-attachments/assets/bb160c1f-21de-4a9a-a1f6-c36372454fb6)

Is it possible to use the uppercase format of the name? For example
```
sketch001 = startSketchOn('XZ')
```

I think we should keep the axis labeling consistent, even if lowercase works?